compute_endpoint/globus_compute_endpoint/engines/high_throughput/worker.py

```diff
@@ -113,85 +112,50 @@ def start(self):
                 result["container_id"] = container_id
                 log.debug("Sending result")
                 # send bytes over the socket back to the manager
                 self.task_socket.send_multipart([b"TASK_RET", dill.dumps(result)])
 
         log.warning("Broke out of the loop... dying")
 
-    def execute_task(self, task_id: str, task_body: bytes) -> dict:
-        log.debug("executing task task_id='%s'", task_id)
-        exec_start = TaskTransition(
-            timestamp=time.time_ns(), state=TaskState.EXEC_START, actor=ActorName.WORKER
+    def compose_exception_message(self, task_id: str) -> bytes:
+        code, user_message = get_result_error_details()
+        outgoing_result = OutgoingResult(
+            task_id=task_id,
+            data=get_error_string(),
+            error_details=OutgoingResultErrorDetails(
+                code=code,
+                user_message=user_message,
+            ),
         )
+        return messagepack.pack(outgoing_result)
 
+    def _worker_execute_task(
+        self, task_id: str, msg: bytes
+    ) -> dict[str, t.Union[str, bytes]]:
+        result_message: dict[str, t.Union[str, bytes]] = {"task_id": task_id}
         try:
-            result = self.call_user_function(task_body)
-        except Exception:
-            log.exception("Caught an exception while executing user function")
-            result_message: dict[
-                str, str | tuple[str, str] | list[TaskTransition]
-            ] = dict(
-                task_id=task_id,
-                exception=get_error_string(),
-                error_details=get_result_error_details(),
-            )
+            # Unwrap HTEX's Task packing
+            task_message = Message.unpack(msg)
+            serialized_fn_package = task_message.task_buffer.decode()
 
-        else:
-            log.debug("Execution completed without exception")
-            result_message = dict(task_id=task_id, data=result)
+            # Deserialize HTEX executors' wrapping of
+            # execute_task, messagepack_payload)
+            function, args, kwargs = self.deserialize(serialized_fn_package)
 
-        exec_end = TaskTransition(
-            timestamp=time.time_ns(),
-            state=TaskState.EXEC_END,
-            actor=ActorName.WORKER,
-        )
+            # Execute
+            serialized_result: bytes = function(*args, **kwargs)
+            result_message["data"] = serialized_result
 
-        result_message["task_statuses"] = [exec_start, exec_end]
+        except Exception:
+            log.exception("Failed to execute task")
+            serialized_error = self.compose_exception_message(task_id)
+            result_message["data"] = serialized_error
 
-        log.debug(
-            "task %s completed in %d ns",
-            task_id,
-            (exec_end.timestamp - exec_start.timestamp),
-        )
         return result_message
 
-    def call_user_function(self, message: bytes) -> str:
```

Comment:
Check whether this method has been migrated into `helper.execute_task`

Reply:
It does not appear to have been.  Here's what I see in `helper.py`:
```python
def execute_task(...):
    ...
    f, args, kwargs = serializer.unpack_and_deserialize(task_buffer)
    result_data = f(*args, **kwargs)
    serialized_data = serializer.serialize(result_data)
```
Note the missing `GC_TASK_TIMEOUT` handling.